pkg/resource/vsc/scan.go

```diff
@@ -0,0 +1,55 @@
+package vsc
+
+import (
+	"errors"
+	"math"
+	"time"
+
+	v1 "github.com/kubernetes-csi/external-snapshotter/client/v6/apis/volumesnapshot/v1"
+
+	"github.com/kyma-project/kyma-metrics-collector/pkg/resource"
+)
+
+const (
+	// storageRoundingFactor rounds of storage to 32. E.g. 17 -> 32, 33 -> 64.
+	storageRoundingFactor = 32
+
+	GiB = 1 << (10 * 3) //nolint:mnd // 1 GiB = 1024^3 bytes
+)
+
+var _ resource.ScanConverter = &Scan{}
+
+type Scan struct {
+	vscs v1.VolumeSnapshotContentList
+}
+
+func (s *Scan) UM(duration time.Duration) (resource.UMMeasurement, error) {
+	return resource.UMMeasurement{}, nil
+}
+
+func (s *Scan) EDP() (resource.EDPMeasurement, error) {
+	edp := resource.EDPMeasurement{}
+
+	var errs []error
+
+	for _, vsc := range s.vscs.Items {
+		if vsc.Status.ReadyToUse != nil && *vsc.Status.ReadyToUse {
+			currVSC := getSizeInGB(*vsc.Status.RestoreSize)
+			edp.ProvisionedVolumes.SizeGbTotal += currVSC
+			edp.ProvisionedVolumes.SizeGbRounded += getVolumeRoundedToFactor(currVSC)
+			edp.ProvisionedVolumes.Count += 1
+		}
+	}
+
+	return edp, errors.Join(errs...)
```

Comment:
currently 
```suggestion
	return edp, nil
```
does the job. Unless we throw an error if restore size is not set